stan/math/fwd/fun/unit_vector_constrain.hpp

```diff
@@ -23,8 +23,8 @@ inline auto unit_vector_constrain(const EigMat& y) {
   plain_type_t<EigMat> unit_vector_y(y_val.size());
   unit_vector_y.val() = unit_vector_constrain(y_val);
 
-  eig_partial squared_norm = dot_self(y_val);
-  eig_partial norm = sqrt(squared_norm);
+  eig_partial norm = norm2(y_val);
+  eig_partial squared_norm = norm * norm;
```

Comment:
These are going to be very similar operationally.  I think the original code may require fewer operations and be more stable arithmetically, because the new one has the sqrt inside norm2, and then you have to the do the additional squaring.  So unless there's something I'm not seeing, I think this should be switched back.  If for some reason you do want to keep the new code, line 27 should be `square(norm)`.